CheckerBase.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+
+namespace Checker
+{
+    public class CheckerBase
+    {
+        public float? MinValue { get; set; }
+
+        public float MaxValue { get; set; }
+
+        public CheckerBase(float minValue, float maxValue)
+        {
+            MinValue = minValue;
+            MaxValue = maxValue;
+        }
+
+        public bool IsValueWithinRange(float value)
+        {
+            return (value < MinValue || value > MaxValue);
+        }
+    }
+
+    public class TemperatureChecker : CheckerBase
+    {
+        public TemperatureChecker() : base(Limits.MIN_TEMPERATURE, Limits.MAX_TEMPERATURE)
+        {
+
+        }
+
+        public new bool IsValueWithinRange(float value)
+        {
+            bool val = base.IsValueWithinRange(value);
+            if (val)
+            {
+                Console.WriteLine("Temperature is out of range!");
+            }
+            return val;
+        }
+    }
+
+    public class SocChecker : CheckerBase
+    {
+        public SocChecker() : base(Limits.MIN_SOC, Limits.MAX_SOC)
```

Comment:
Hi, it was intended to do so because if we have some new features added with respect to temperature check or the soc checker we could use the extended classes. Because in future if we have new features added to those cases, at that time we may not want to again refactor the base logic.

Reply:
Kindly suggest which would be the better approach